pkgs/development/compilers/ghc/with-packages.nix

```diff
@@ -44,6 +47,7 @@ buildEnv {
       rm -f $out/bin/$prg
       makeWrapper ${ghc}/bin/$prg $out/bin/$prg         \
         --add-flags '"-B$NIX_GHC_LIBDIR"'               \
+        ${optionalString exposeGHC "--add-flags '-package ghc'"} \
```

Comment:
@gridaphobe, if you want to run `ghc-pkg expose ghc` in the `postInstall` hook, then you must first make sure that `$out/lib/ghc-${ghc.version}/package.conf.d` is a writable directory. This should be always the case when you include at least one other Haskell packages other than `ghc` itself in the `ghcWithPackages` environment. The following example works fine for me. I define:

    # ~/.nixpkgs/config.nix

    {
      packageOverrides = super: let self = super.pkgs; in
      {
        ghcDevEnv = (self.haskellPackages.ghcWithPackages (p: with p; [stm])).overrideDerivation (drv: { postBuild = ''
          ${drv.postBuild}
          $out/bin/ghc-pkg expose ghc
        ''; });
      };
    }

and then run:

    $ nix-build -o /tmp/haskell ~/.nix-defexpr --option binary-caches http://hydra.cryp.to -A ghcDevEnv
    these derivations will be built:
      /nix/store/dq4g8w491yj6qblzvscgv2i4fvmdv5a0-haskell-env-ghc-7.8.4.drv
    building path(s) ‘/nix/store/0i532km3pj8b5n87nd9b267zci1mkjw6-haskell-env-ghc-7.8.4’
    building /nix/store/0i532km3pj8b5n87nd9b267zci1mkjw6-haskell-env-ghc-7.8.4
    created 90 symlinks in user environment
    /nix/store/0i532km3pj8b5n87nd9b267zci1mkjw6-haskell-env-ghc-7.8.4

    $ /tmp/haskell/bin/ghci
    GHCi, version 7.8.4: http://www.haskell.org/ghc/  :? for help
    Loading package ghc-prim ... linking ... done.
    Loading package integer-gmp ... linking ... done.
    Loading package base ... linking ... done.
    Prelude> import GHC
    Prelude GHC> :q
    Leaving GHCi.

Works like a charm!